src/pages/saksbehandling/avsluttBehandling/lukkSøknad/LukkSøknad.tsx

```diff
@@ -0,0 +1,230 @@
+import * as Routes from '~lib/routes';
+import { LukkSøknadBegrunnelse, Søknad, Søknadstype } from '~types/Søknad';
+
+import AvslåttSøknad from '../../avslag/AvslåttSøknad';
+import AvsluttBehandlingBunnknapper from '../avsluttBehandlingBunnknapper/AvsluttBehandlingBunnknapper';
+
+import Avvist from './Avvist';
+import nb from './lukkSøknad-nb';
+import styles from './lukkSøknad.module.less';
+import {
+    lukkSøknadInitialValues,
+    LukkSøknadOgAvsluttSøknadsbehandlingFormData,
+    AvsluttSøknadsbehandlingBegrunnelse,
+    LukkSøknadOgAvsluttSøknadsbehandlingType,
+    getLukkSøknadValidationSchema,
+} from './lukkSøknadUtils';
+import Trukket from './Trukket';
+
+const LukkSøknadOgAvsluttBehandling = (props: { sakId: string; søknad: Søknad }) => {
+    const history = useHistory();
+    const { formatMessage } = useI18n({ messages: nb });
+    const [søknadLukketStatus, lukkSøknad, resetLukkSøknadStatus] = useAsyncActionCreator(sakSlice.lukkSøknad);
+    const [avslagManglendeDokStatus, avslåPgaManglendeDok, resetAvslagManglendeDokStatus] = useAsyncActionCreator(
+        sakSlice.avslagManglendeDokSøknad
+    );
+
+    const submitStatus = pickRemoteData(søknadLukketStatus, avslagManglendeDokStatus);
+
+    const handleSubmit = async (values: LukkSøknadOgAvsluttSøknadsbehandlingFormData) => {
+        console.log('handle submit');
+        if (!values.begrunnelse) {
+            return;
+        }
+
+        if (values.begrunnelse === AvsluttSøknadsbehandlingBegrunnelse.ManglendeDok) {
+            avslåPgaManglendeDok(
+                {
+                    søknadId: props.søknad.id,
+                    //validering fanger denne
+                    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                    body: { fritekst: values.manglendeDok.fritekst! },
+                },
+                () => {
+                    const message = formatMessage('avslutt.behandlingHarBlittAvsluttet');
+                    return history.push(Routes.createSakIntroLocation(message, props.sakId));
+                }
+            );
+        } else {
+            lukkSøknad(
+                {
+                    søknadId: props.søknad.id,
+                    body: lagBody(values),
+                },
+                () => {
+                    const message = formatMessage('avslutt.behandlingHarBlittAvsluttet');
+                    history.push(Routes.createSakIntroLocation(message, props.sakId));
+                }
+            );
+        }
+    };
+
+    const form = useForm<LukkSøknadOgAvsluttSøknadsbehandlingFormData>({
+        defaultValues: lukkSøknadInitialValues,
+        resolver(values, ...args) {
+            return yupResolver(getLukkSøknadValidationSchema(values.begrunnelse))(values, ...args);
+        },
+    });
+
+    const watchBegrunnelse = form.watch('begrunnelse');
+
+    const handleRequestValidate = (onSuccess: () => void): void => {
+        form.handleSubmit(onSuccess)();
+    };
```

Comment:
Dette er vel det samme som
```
const handleRequestValidate = (onSuccess: () => void) => form.handleSubmit(onSuccess)
```

som igjen da bare kan skrives `form.handleSubmit` hvor den brukes? Da slipper vi en ekstra høyere-ordens-funksjon? Eller er det noe jeg ikke catcher? 😄  Hvis ikke er det kanskje ikke behov for den hjelpemetoden i det hele tatt?